### PR TITLE
Change AwsLambdaInstrumentor span attrs to meet faas semconv

### LINKED ISSUE #2371: Change AwsLambdaInstrumentor span attribute settings to match faas semconv
**Is your feature request related to a problem?**
No.

**Describe the solution you'd like**
Python AwsLambdaInstrumentor attribute settings are not consistent with the [latest faas semconv](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/faas/aws-lambda.md#all-triggers). Specifically, I think these instrumentor changes are needed:
1. `ResourceAttributes.FAAS_ID` --> `ResourceAttributes.CLOUD_RESOURCE_ID`
2. `SpanAttributes.FAAS_EXECUTION` --> `SpanAttributes.FAAS_INVOCATION_ID`

For this solution, change [these set_attribute calls](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/37aba928d45713842941c7efc992726a79ea7d8a/instrumentation/opentelemetry-instrumentation-aws-lambda/src/opentelemetry/instrumentation/aws_lambda/__init__.py#L342-L355). Make special note in the release notes that this is breaking. This is not additive like #2368.

**Describe alternatives you've considered**
I've thought about and brought up 2 other ideas on Slack:

Idea (A)
The Python AwsLambdaInstrumentor upgrades to a newer semconv version. Would it need to be an opt-in for users like for http instrumentation? (i.e. [this recent issue](https://github.com/open-telemetry/opentelemetry-python-contrib/issues/2351)) Would this need to be presented to an Otel Spec project group? (i.e. [this issue](https://github.com/open-telemetry/opentelemetry-specification/issues/3042))

Idea (B)
The Python AwsLambdaInstrumentor gets request/response hooks implemented like in NodeJS. This would allow users to add any attributes they like. I see this was [requested](https://github.com/open-telemetry/opentelemetry-python-contrib/issues/1140), [discussed in Otel Spec](https://github.com/open-telemetry/opentelemetry-specification/issues/3042#issuecomment-1384964399), [PR'd](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1476) (rejected), discussed at [faas project sig](https://docs.google.com/document/d/187XYoQcXQ9JxS_5v2wvZ0NEysaJ02xoOYNXj08pT0zc/edit#heading=h.w0bc5a8eac74), alternatively [PR'd as a SpanProcessor requirement](https://github.com/open-telemetry/opentelemetry-specification/pull/3189) (rejected), and brought up again in [Slack](https://cloud-native.slack.com/archives/C04HVBETC9Z/p1684317968674529). I haven't found any other updates since May 2023.

In [this otel-faas thread](https://cloud-native.slack.com/archives/C04HVBETC9Z/p1711387797268909), it was mentioned that the faas attributes are not stable so a breaking change would be ok. Therefore we could implement the quick, in-place change.

**Additional context**
1. https://cloud-native.slack.com/archives/C01PD4HUVBL/p1711140906040599
2. https://cloud-native.slack.com/archives/C04HVBETC9Z/p1711387797268909

### PR DESCRIPTION
**Pull request by tammy-baylis-swi**
# Description

Python AwsLambdaInstrumentor attribute settings are not consistent with the [latest faas semconv](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/faas/aws-lambda.md#all-triggers). Specifically, I think these instrumentor changes are needed which are updated in this PR:

1. `ResourceAttributes.FAAS_ID` --> `ResourceAttributes.CLOUD_RESOURCE_ID`
2. `SpanAttributes.FAAS_EXECUTION` --> `SpanAttributes.FAAS_INVOCATION_ID`

Looking at the semconvgen on main, `faas.id` and `faas.execution` are deprecated and should be replaced by `cloud.resource_id` and `faas.invocation_id`, respectively (described [here](https://github.com/open-telemetry/opentelemetry-python/blob/29c293fbb120ff8b760519a25e0530b5ab05af83/opentelemetry-semantic-conventions/src/opentelemetry/semconv/resource/__init__.py#L647-L650) and [here](https://github.com/open-telemetry/opentelemetry-python/blob/29c293fbb120ff8b760519a25e0530b5ab05af83/opentelemetry-semantic-conventions/src/opentelemetry/semconv/trace/__init__.py#L1321-L1324)).

Fixes #2371 

## Type of change

Please delete options that are not relevant.

- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A

# Does This PR Require a Core Repo Change?

- [X] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [X] Followed the style guidelines of this project
- [x] Changelogs have been updated
- [X] Unit tests have been added
- [ ] Documentation has been updated

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -27,6 +27,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   ([#2355](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2355))
 - AwsLambdaInstrumentor sets `cloud.account.id` span attribute
   ([#2367](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2367))
+- AwsLambdaInstrumentor sets `cloud.resource_id`, `faas.invocation_id` span attributes (breaking)
```
Comment (lzchen) [inline CHANGELOG.md:30]: Could you add a `### Breaking changes` section in the CHANGELOG so we can capture these semver changes altogether?

Comment (lzchen) [reply CHANGELOG.md:30]: Actually it is done in this PR

Comment (tammy-baylis-swi) [reply CHANGELOG.md:30]: Sure will do!

**Review by lzchen — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (xrmx) [reply CHANGELOG.md:39]: @tammy-baylis-swi we have the breaking changes section in main, could you please move it there please?

Comment (tammy-baylis-swi) [reply CHANGELOG.md]: Great! Moved and improved wording in [50b446e](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2372/commits/50b446e23d132794934bbba0138aa4dc413c90c6)

**Review by xrmx — APPROVED** [review]

**Pull request merged**